### ISSUE TITLE
INCLUDE_BACKGROUND is confusing 

### ISSUE THREAD

**Issue by gattia** [ad12/MedSegPy#31]
I think walking through the below I figured out what is going on - but this was confusing to me, and in the end, I don't think it allows the user to include a background class in their loss function during training (which they might want). Im also wondering if there is an implication for using the `Softmax` loss. 

In config.py `INCLUDE_BACKGROUND` default is `False` and indicates if `True` should include a background mask, if `False` shouldn't:
https://github.com/ad12/MedSegPy/blob/0c316baaf040c22d562940a198a0e48eef2d36a8/medsegpy/config.py#L623-L625

in losses.py `INCLUDE_BACKGROUND` is passed directly to `get_training_loss` for the parameter `remove_background`: 
https://github.com/ad12/MedSegPy/blob/0c316baaf040c22d562940a198a0e48eef2d36a8/medsegpy/losses.py#L65-L71

This seems backwards based on the names alone, and is confirmed looking at the downstream class `DiceLoss` where `remove_background` removes the background label. So, if `INCLUDE_BACKGROUND=True` it will have `remove_background=True` which means we'll actually end up removing the background and not including it?
https://github.com/ad12/MedSegPy/blob/0c316baaf040c22d562940a198a0e48eef2d36a8/medsegpy/loss/classification.py#L78-L79 
 

But looking through other files `INCLUDE_BACKGROUND` seems to be used in different ways. In [im_gens.py](https://github.com/ad12/MedSegPy/blob/master/medsegpy/data/im_gens.py) it seems to be used properly to add a background class to the array. 

Ok, here's where I think I started to understand?  `sem_seg_evaluation.py` & `oai_test.py` both use it to determine if they should remove the background: 
https://github.com/ad12/MedSegPy/blob/0c316baaf040c22d562940a198a0e48eef2d36a8/medsegpy/evaluation/sem_seg_evaluation.py#L133-L137

https://github.com/ad12/MedSegPy/blob/0c316baaf040c22d562940a198a0e48eef2d36a8/projects/TechConsiderations/oai_test.py#L208-L211

They both also include a comment saying "background is always excluded from analysis". So, is `INCLUDE_BACKGROUND` being passed to `remove_background` in `DiceLoss` because we always want to skip/not consider the background class for the loss function, and we're just telling the loss function if it exists in the data so it should be removed? 

Im guessing this might have something to do with using the softmax function, otherwise Im not clear on why we'd add a background to just exclude it from the loss. Whether this is the case or not, then I'm guessing that if the activation is a softmax we should be turning include_background to be `True` - this might be happening somewhere but I couldn't find it. If this is the case and it isn't being done somewhere, this could also explain some of the discrepancy between using `sigmoid` and `softmax` that I've experienced in training, because from `config.py` it looks like the default is to exclude the background: 
https://github.com/ad12/MedSegPy/blob/0c316baaf040c22d562940a198a0e48eef2d36a8/medsegpy/config.py#L116
